sklearn/neighbors/base.py

```diff
@@ -359,11 +360,22 @@ class from an array representing our data set and ask who's
                     X, self._fit_X, self.effective_metric_, n_jobs=n_jobs,
                     **self.effective_metric_params_)
 
-            neigh_ind = np.argpartition(dist, n_neighbors - 1, axis=1)
-            neigh_ind = neigh_ind[:, :n_neighbors]
-            # argpartition doesn't guarantee sorted order, so we sort again
-            neigh_ind = neigh_ind[
-                sample_range, np.argsort(dist[sample_range, neigh_ind])]
+            if issparse(dist):
+                if np.any(getnnz(dist, axis=1) < n_neighbors):
+                    raise ValueError("Not enough neighbors in sparse "
+                                     "precomputed matrix to get {} "
+                                     "nearest neighbors".format(n_neighbors))
+                neigh_ind = np.zeros((dist.shape[0], n_neighbors))
+                for i in range(0, dist.shape[0]):
+                    row = dist.getrow(i)
```

Comment:
This may be unnecessarily expensive computationally. Perhaps it deserves some benchmarking: Instead of `row.indices`, use `dist.indices[dist.indptr[i]:dist.indptr[i+1]]` and similar for `data`.

Reply:
Done. But how do I break this statement into multiple lines to avoid pep8 failure? :)